tests/codegen/intrinsics/transmute-x64.rs

```diff
@@ -9,9 +9,10 @@ use std::mem::transmute;
 // CHECK-LABEL: @check_sse_float_to_int(
 #[no_mangle]
 pub unsafe fn check_sse_float_to_int(x: __m128) -> __m128i {
-    // CHECK-NOT: alloca
-    // CHECK: %0 = load <4 x float>, ptr %x, align 16
-    // CHECK: store <4 x float> %0, ptr %_0, align 16
+    // FIXME: the MIR opt still works, but the ABI logic now introduces
+    // an alloca here.
+    // CHECK: alloca
+    // CHECK: store <4 x float> %x, ptr %_0, align 16
```

Comment:
I have no idea what this was trying to test, but it probably doesn't test that any more. The alloca is emitted by the ABI handling, and this test disables LLVM optimizations, so there's no way we can avoid the alloca.

Cc @scottmcm (who added the test in https://github.com/rust-lang/rust/commit/d757c4b904869967f1e665dc2bb9a2ca5122bc96)

Reply:
This is a regression test for an ICE in cg_ssa: https://github.com/rust-lang/rust/pull/110021/commits/d757c4b904869967f1e665dc2bb9a2ca5122bc96

Reply:
But why would that be a codegen test...?

Reply:
hmm @scottmcm can you explain?

Reply:
So should I remove the test then? Or is there some way to still test what actually matters here?

Reply:
@scottmcm So should I remove the test then? Or is there some way to still test what actually matters here?

Reply:
I ended up removing the test; I can't figure out how to write a test that does not contain `alloca`. We always emit `alloca` for the implicit transmutes caused by the ABI, so there's no way to reasonably test for their absence in a test that disables LLVM opts.

@scottmcm is it worth opening an issue to track improving our codegen for the implicit ABI transmutes? It would maybe use some of the same optimizations as explicit transmutes... or maybe not, I have no idea.